test/ng/directive/inputSpec.js

```diff
@@ -1879,6 +1966,30 @@ describe('input', function() {
       dealoc(formElm);
     });
 
+    it('should not reuse the hour part of a previous date object after changing the timezone', function() {
```

Comment:
Not sure how this test checks that the hour part of the previous date object isn't used.

Reply:
Also Revised this test and added more comments.